powerauth-nextstep-model/src/main/java/io/getlime/security/powerauth/lib/nextstep/model/entity/OperationFormData.java

```diff
@@ -12,133 +15,246 @@
+     * @param noteId Note ID.
      */
-    public boolean isDynamicDataLoaded() {
-        return dynamicDataLoaded;
+    @JsonIgnore
+    public void setNote(String noteId) {
+        OperationNoteAttribute attr = new OperationNoteAttribute();
+        attr.setId(noteId);
+        saveAttribute(attr);
     }
 
+
     /**
-     * Set whether dynamic data is loaded.
-     * @param dynamicDataLoaded True if dynamic data is loaded, otherwise false.
+     * Set localized note.
+     * @param noteId Note ID.
+     * @param note Localized note.
      */
-    public void setDynamicDataLoaded(boolean dynamicDataLoaded) {
-        this.dynamicDataLoaded = dynamicDataLoaded;
+    @JsonIgnore
+    public void setNote(String noteId, String note) {
+        OperationNoteAttribute attr = new OperationNoteAttribute();
+        attr.setId(noteId);
+        attr.setMessage(note);
+        saveAttribute(attr);
     }
 
     /**
-     * Get key-value map with user input.
-     * @return User input.
+     * Get note.
+     * @return Note.
      */
-    public Map<String, String> getUserInput() {
-        return userInput;
+    @JsonIgnore
+    public OperationNoteAttribute getNote() {
+        List<OperationFormAttribute> attrs = getAttributesByType(OperationFormAttribute.Type.MESSAGE);
+        if (attrs == null) {
+            return null;
+        }
+        if (attrs.size()>1) {
+            throw new IllegalStateException("Multiple attributes of type MESSAGE found");
+        }
+        return (OperationNoteAttribute) attrs.get(0);
     }
 
     /**
-     * Add an amount.
-     * @param amountLabel Amount label.
-     * @param currencyLabel Currency label.
-     * @param amount Amount value.
-     * @param currency Amount currency.
+     * Add a bank account choice.
+     * @param id Bank account choice ID.
+     * @param bankAccounts List of bank accounts.
      */
-    public void addAmount(String amountLabel, String currencyLabel, BigDecimal amount, String currency) {
-        OperationAmountAttribute attr = new OperationAmountAttribute();
-        attr.setLabel(amountLabel);
-        attr.setCurrencyLabel(currencyLabel);
-        attr.setAmount(amount);
-        attr.setCurrency(currency);
-        parameters.add(attr);
+    @JsonIgnore
+    public void addBankAccountChoice(String id, List<BankAccountDetail> bankAccounts) {
+        OperationBankAccountChoiceAttribute attr = new OperationBankAccountChoiceAttribute();
+        attr.setId(id);
+        attr.setBankAccounts(bankAccounts);
+        saveAttribute(attr);
     }
 
     /**
      * Add a key-value attribute.
-     * @param label Attribute label.
+     * @param id Attribute ID.
      * @param value Attribute value.
      */
-    public void addKeyValue(String label, String value) {
+    @JsonIgnore
+    public void addKeyValue(String id, String value) {
         OperationKeyValueAttribute attr = new OperationKeyValueAttribute();
-        attr.setLabel(label);
+        attr.setId(id);
```

Comment:
I don't get it - we still need "label", in order to be able to localize the label, right? We cannot do it by ID, since it is a lookup / addressing identifier that has nothing to do with localization...

Reply:
As discussed below, I made label a generic field. Then by calling:

String localizedValue = localize(id);
attribute.setLabel(localizedValue);

we can simply localize all OperationFormAttributes.

But if you think it does not make sense to have labels for some OperationFormAttributes, method setLabel() can be called depending on actual availability of the label.

Reply:
OK - true. Your way is better.